pallets/kilt-launch/src/lib.rs

```diff
@@ -0,0 +1,797 @@
+//!   an unowned account to a user-owned account. Requires signature of a
+//!   special account `TransferAccount` which does not have any other super
+//!   powers.
+//! - `migrate_multiple_genesis_accounts` - Migrate vesting or the KILT balance
+//!   lock from a list of unowned accounts to the same target user-owned
+//!   account. Requires signature of a special account `TransferAccount` which
+//!   does not have any other super powers.
+//! - `locked_transfer` - Transfer locked tokens from one user-owned account to
+//!   another user-owned account. This will be made available shortly before
+//!   removing the sudo key because the purpose of the lock to disable
+//!   transferability of the amount.
+//! - `change_transfer_account` - Change the transfer account. Can only be
+//!   called by sudo.
+//! - `force_unlock` - Remove all locks for a given block. Can only be called by
+//!   sudo.
+//!
+//! ## Genesis config
+//!
+//! The KiltLaunch pallet depends on the [`GenesisConfig`].
+//!
+//! ## Assumptions
+//!
+//! * All accounts provided with balance and potentially vesting or a KILT
+//!   balance lock in the genesis block are not owned by anyone and have to be
+//!   migrated to accounts which are owned by users.
+//! * All unowned accounts have either vesting, the KILT balance lock or neither
+//!   of both. This assumption is neither checked, nor forced, nor does any code
+//!   break if it does not hold true.
+//! * Vesting starts at genesis block for all unowned addresses which should be
+//!   migrated to user-owned accounts. This assumption is checked during
+//!   migration.
+//! * All KILT balance locks end at the same block for all unowned addresses
+//!   which should be migrated to user-owned accounts. This assumption is
+//!   checked during migration and locked transfer.
+//! * The total number of accounts for which a KILT balance lock is set up is at
+//!   most `MaxClaims`, for us it will be ~6. This assumption is not checked
+//!   when appending to `UnlockedAt`.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+#![allow(clippy::unused_unit)]
+
+pub use default_weights::WeightInfo;
+#[cfg(feature = "std")]
+use frame_support::traits::GenesisBuild;
+use frame_support::{
+	dispatch::DispatchResultWithPostInfo,
+	inherent::Vec,
+	pallet_prelude::*,
+	sp_runtime::traits::{StaticLookup, Zero},
+	storage::types::StorageMap,
+	traits::{Currency, ExistenceRequirement::AllowDeath, Get, LockIdentifier, LockableCurrency, WithdrawReasons},
+	transactional,
+};
+pub use pallet::*;
+use pallet_balances::{BalanceLock, Locks};
+use pallet_vesting::{Vesting, VestingInfo};
+use sp_runtime::traits::{Convert, Saturating};
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+#[cfg(feature = "runtime-benchmarks")]
+pub mod benchmarking;
+pub mod default_weights;
+
+pub const KILT_LAUNCH_ID: LockIdentifier = *b"kiltlnch";
+pub const VESTING_ID: LockIdentifier = *b"vesting ";
```

Comment:
It is intentional and we have to keep it to not cause trouble with utility from the vesting pallet. It is definitely hacky but the only other solution I see is to use the vesting pallet from a local copy rather than a dependency and then make the id public. Would you rather do it that way?

Reply:
As discussed: We will not add the `pallet_vesting` manually. Our unit tests will show whenever they change the `LockIdentifier` since we rely on functionality from `pallet_vesting` to update and eventually remove the lock.